src/transaction/log_manager.c

```diff
@@ -14684,6 +14733,457 @@ cdc_is_filtered_user (char *user)
+  LOG_LSA next_log_rec_lsa = LSA_INITIALIZER;
+  LOG_LSA process_lsa = LSA_INITIALIZER;
+
+  LOG_PAGE *log_page_p = NULL;
+  char log_pgbuf[IO_MAX_PAGE_SIZE + MAX_ALIGNMENT];
+
+  LOG_RECORD_HEADER *log_rec_header;
+
+  char tran_user[DB_MAX_USER_LENGTH + 1] = "\0";
+  int trid;
+
+  int error = NO_ERROR;
+
+  LOG_RECTYPE log_type;
+
+  char *supplement_data = NULL;
+
+  RECDES supp_recdes = RECDES_INITIALIZER;
+  RECDES undo_recdes = RECDES_INITIALIZER;
+  RECDES redo_recdes = RECDES_INITIALIZER;
+
+  int num_loginfo = 0;
+  CDC_LOGINFO_ENTRY *log_info_entry = NULL;
+
+  if (LSA_ISNULL (&context->start_lsa))
+    {
+      if ((error = flashback_find_start_lsa (thread_p, context)) != NO_ERROR)
+	{
+	  goto exit;
+	}
+    }
+
+  if (context->forward)
+    {
+      LSA_COPY (&process_lsa, &context->start_lsa);
+    }
+  else
+    {
+      LSA_COPY (&process_lsa, &context->end_lsa);
+    }
+
+  log_page_p = (LOG_PAGE *) PTR_ALIGN (log_pgbuf, MAX_ALIGNMENT);
+
+  /* fetch log page */
+  if (logpb_fetch_page (thread_p, &process_lsa, LOG_CS_SAFE_READER, log_page_p) != NO_ERROR)
+    {
+      error = ER_FAILED;
+      goto exit;
+    }
+
+begin:
+
+  LSA_COPY (&cur_log_rec_lsa, &process_lsa);
+
+  log_rec_header = LOG_GET_LOG_RECORD_HEADER (log_page_p, &process_lsa);
+
+  log_type = log_rec_header->type;
+  trid = log_rec_header->trid;
+
+  if (context->forward)
+    {
+      if (LSA_ISNULL (&log_rec_header->forw_lsa) && log_type != LOG_END_OF_LOG)
+	{
+	  if (log_startof_nxrec (thread_p, &next_log_rec_lsa, false) == NULL)
+	    {
+	      /* can not get next log record lsa */
+	      error = ER_FAILED;
+	      goto exit;
+	    }
+	}
+      else
+	{
+	  LSA_COPY (&next_log_rec_lsa, &log_rec_header->forw_lsa);
+	}
+
+      if (LSA_ISNULL (&next_log_rec_lsa) && logpb_is_page_in_archive (cur_log_rec_lsa.pageid))
+	{
+	  next_log_rec_lsa.pageid = cur_log_rec_lsa.pageid + 1;
+	}
+    }
+  else
+    {
+      LSA_COPY (&next_log_rec_lsa, &log_rec_header->prev_tranlsa);
+    }
+
+  LOG_READ_ADD_ALIGN (thread_p, sizeof (*log_rec_header), &process_lsa, log_page_p);
+
+  if (trid == context->trid && log_type == LOG_SUPPLEMENTAL_INFO)
+    {
+      /*supplemental log info types : time, tran_user, undo image */
+      LOG_REC_SUPPLEMENT *supplement;
+      int supplement_length;
+      SUPPLEMENT_REC_TYPE rec_type;
+
+      OID classoid;
+
+      LOG_LSA undo_lsa, redo_lsa;
+
+      LOG_READ_ADVANCE_WHEN_DOESNT_FIT (thread_p, sizeof (*supplement), &process_lsa, log_page_p);
+
+      supplement = (LOG_REC_SUPPLEMENT *) (log_page_p->area + process_lsa.offset);
+      supplement_length = supplement->length;
+      rec_type = supplement->rec_type;
+
+      LOG_READ_ADD_ALIGN (thread_p, sizeof (*supplement), &process_lsa, log_page_p);
+
+      if (cdc_get_undo_record (thread_p, log_page_p, cur_log_rec_lsa, &supp_recdes) != S_SUCCESS)
+	{
+	  error = ER_FAILED;
+	  goto exit;
+	}
+
+      supplement_length = sizeof (supp_recdes.type) + supp_recdes.length;
```

Comment:
These codes around this line are very similar to the codes in the flashback_make_summary_list function (case LOG_SUPPLEMENTAL_INFO:). So, make a function for them.

Reply:
Yes, it will be better

Reply:
OK.